zerver/views/auth.py

```diff
@@ -757,6 +757,10 @@ def login_page(
         # We want to authenticate user while letting to be logged in.
         is_preview = True
 
+    if 'action_key' in request.POST:
+        is_preview = True
+        extra_context['action_key'] = request.POST['action_key']
```

Comment:
This code is for a special case i.e If the user is logged in, and clicks the link to re-authenticate then if for some reason the user gets a validation error we generally redirect to \login with some form error. The redirect takes user to home page as the user is already loggeed in. So, this code takes care of restricting that redirect to home page